server/app/views/admin/programs/ProgramBlockView.java

```diff
@@ -42,4 +51,48 @@ protected final DivTag renderProgramInfo(ProgramDefinition programDefinition) {
             editDetailsButton)
         .withClasses("bg-gray-100", "text-gray-800", "shadow-md", "p-8", "pt-4", "-mx-2");
   }
+
+  /** Renders a div presenting the predicate definition for the admin. */
+  protected final DivTag renderExistingPredicate(
+      String blockName,
+      PredicateDefinition predicateDefinition,
+      ImmutableList<QuestionDefinition> questionDefinitions) {
+    DivTag container = div();
+
+    if (predicateDefinition
+        .computePredicateFormat()
+        .equals(PredicateDefinition.PredicateFormat.SINGLE_QUESTION)) {
+      return container.with(
+          text(predicateDefinition.toDisplayString(blockName, questionDefinitions)));
+    } else if (!predicateDefinition
+        .computePredicateFormat()
+        .equals(PredicateDefinition.PredicateFormat.OR_OF_SINGLE_LAYER_ANDS)) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Predicate type %s is unsupported.", predicateDefinition.computePredicateFormat()));
+    }
+
+    ImmutableList<PredicateExpressionNode> andNodes =
+        predicateDefinition.rootNode().getOrNode().children();
+
+    if (andNodes.size() == 1) {
+      return container.with(
+          text(
+              blockName
+                  + " is "
+                  + predicateDefinition.action().toDisplayString()
+                  + " "
```

Comment:
should there be a joiner word here and below like " if " given there's some sentence structure.

HIDDEN if XXX  

SHOWN if any of: XXXX

Reply:
The predicate action toString includes the "if"

https://github.com/civiform/civiform/blob/main/server/app/services/program/predicate/PredicateAction.java#L8